src/layouts/Settings/ColourSettings.tsx

```diff
@@ -0,0 +1,148 @@
+}
+
+export const ColourSettings = ({
+  isError,
+}: ColourSettingsProp): JSX.Element => {
+  const { register } = useFormContext()
+  return (
+    <Section id="colour-fields">
+      <SectionHeader label="Colours" />
+      <VStack spacing="1rem" align="flex-start" w="50%">
+        <FormColourPicker
+          {...register("primaryColour")}
+          label="Primary"
+          isDisabled={isError}
+        />
+        <FormColourPicker
+          {...register("secondaryColour")}
+          label="Secondary"
+          isDisabled={isError}
+        />
+        <FormColourPicker
+          {...register("mediaColours.0")}
+          label="Resource 1"
+          isDisabled={isError}
+        />
+        <FormColourPicker
+          {...register("mediaColours.1")}
+          label="Resource 2"
+          isDisabled={isError}
+        />
+        <FormColourPicker
+          {...register("mediaColours.2")}
+          label="Resource 2"
+          isDisabled={isError}
+        />
+        <FormColourPicker
+          {...register("mediaColours.3")}
+          label="Resource 3"
+          isDisabled={isError}
+        />
+        <FormColourPicker
+          {...register("mediaColours.4")}
+          label="Resource 4"
+          isDisabled={isError}
+        />
+      </VStack>
+    </Section>
+  )
+}
+
+interface FormColourPickerProps extends InputProps {
+  name: string
+  label: string
+  isDisabled?: boolean
+}
+
+const FormColourPicker = forwardRef<FormColourPickerProps, "input">(
+  (
+    { name, label, isDisabled, ...rest }: FormColourPickerProps,
+    ref
+  ): JSX.Element => {
+    const fieldName = `colours.${name}`
+    const { isOpen, onOpen, onClose } = useDisclosure()
+    const { setValue, getValues } = useFormContext()
+    const initialColour: string = getValues(fieldName)
+    const [colour, setColour] = useState<string>(initialColour)
+
+    return (
+      <>
+        <HStack justify="flex-start" w="100%" spacing="2rem">
+          <FormLabel w="25%">{upperFirst(label)}</FormLabel>
+          <Flex>
+            <Input
+              disabled
+              {...rest}
+              value={initialColour}
+              borderRightRadius={0}
+            />
+            <Button
+              borderLeftRadius={0}
+              isDisabled={isDisabled}
+              _hover={{
+                backgroundColor: varyHexColor(initialColour, 10),
+              }}
+              _active={{
+                backgroundColor: varyHexColor(initialColour, 20),
+              }}
+              aria-label="Select colour"
+              bgColor={colour}
+              onClick={onOpen}
+            />
+          </Flex>
+        </HStack>
+        <Modal onClose={onClose} isOpen={isOpen} isCentered>
+          <ModalContent w="auto">
+            <SketchPicker
+              width="250px"
+              color={colour}
+              disableAlpha
+              onChange={(colourResult) => setColour(colourResult.hex)}
+            />
```

Comment:
I think we might need to set an onblur to return to the original colour - currently we face the following issue if the user clicks out of the colour picker:

https://user-images.githubusercontent.com/22111124/173776874-aaf758a3-7498-4787-b5df-431d1fabaf3c.mov

Reply:
wow dang, this is a good catch! 

> The refactored settings look great! Left some comments.
> 
> Also, we might want to remove the `(optional)` after every field, or at least particularly for the toggles

will double check w/ designers on this and update accordingly!